content/en/blog/postman-grpcweb.md

```diff
@@ -0,0 +1,328 @@
+    language-agnostic API contract. This can be used to generate client and
+    server code in multiple languages, providing a seamless experience for
+    developers.
+
+
+## Setting Up a gRPC-Web Environment
+
+Getting started with gRPC-Web will require defining services and message
+payloads using Protobuf, setting up the gRPC backend service (or [a mock server
+for the time being](https://blog.postman.com/postman-mocking-magic-for-grpcs/)),
+and configuring an Envoy proxy to translate between gRPC-Web and gRPC.
+
+First, you define your service in a .proto file:
+
+```proto
+syntax = "proto3";
+
+package example;
+
+service ExampleService {
+  rpc GetExampleData(ExampleRequest) returns (ExampleResponse);
+}
+
+message ExampleRequest {
+  string query = 1;
+}
+
+message ExampleResponse {
+  repeated string data = 1;
+}
+```
+
+This `.proto` file defines a simple service with a single RPC method,
+`GetExampleData`, along with the request and response message formats. Since the
+operation sends a single `ExampleRequest` message in the request, and expects to
+receive a single `ExampleResponse` message in the response, this unary RPC call
+mimics a RESTful request.
+
+Next, generate the client stub code for your service using the protoc
+command-line tool with the appropriate gRPC-Web plugin. ([Here is an
+example](https://blog.postman.com/postman-mocking-magic-for-grpcs/) from the
+gRPC-Web quickstart documentation). This process will create the JavaScript
+client files you'll need to make gRPC-Web calls from the browser.
+
+After your gRPC server is implemented in the language of your choice, you'll
+configure an Envoy proxy. [Here is another
+example](https://blog.postman.com/postman-mocking-magic-for-grpcs/) from the
+gRPC-Web quickstart documentation.
+
+Here is some YAML syntax of an Envoy configuration which enables gRPC-Web as
+part of the larger configuration linked above.
+
+```yaml
+http_filters:
+- name: envoy.filters.http.grpc_web
+- name: envoy.filters.http.router
+```
+
+With these elements in place, you can start making gRPC-Web calls from your web
+application.
+
+
+## Defining Service Methods with Protobuf
+
+When defining your service methods, Protobuf acts as the single source of truth
+by defining the message structure for requests and responses. This strict schema
+allows for automatic client and server code generation in multiple languages.
+For JavaScript in particular, this code generation streamlines the call process
+for the browser client.
+
+Using the example .proto file above, the generated JavaScript client code would
+use these definitions to ensure that only the correct data types are sent and
+received. This process handles much of the manual data validation and parsing
+that can be error-prone with RESTful services.
+
+
+## Replacing a Typical WebSocket Connection with gRPC-Web
+
+WebSockets provide a full-duplex communication channel over a single long-lived
+connection. There are technical limitations of
```

Comment:
> There are technical limitations of [WebTransport](https://developer.chrome.com/articles/webtransport/) in browsers which currently block client streaming over HTTP/2.

I'm not exactly sure what's the purpose of this sentence here (i missed it during initial review), since grpc-web is not YET on WebTransport (although it would be our long-term plan to do so).

Also, this statement is rather vague as it's not quite clear what the limitations are :)

Reply:
I'm open to ideas on improving this. I was attempting to show that grpc-web is probably not a drop in replacement, currently, for WebSockets because of the limitations from browsers.

Reply:
Could you maybe follow up in the original doc we had? @wenbozhu had 2 open comments there too. One about this particular thread.

Reply:
I cannot, as I no longer work at Postman.

Reply:
Oh ok!! Sorry i didn't know..

Quoting from the doc comment:
_With WebTransport, you don't really need HTTP/2 (besides as the fallback when QUIC is blocked)_

---

My personal opinion is to maybe consider removing this sentence, because WebTransport is not yet part of grpc-web and hence might be irrelevant for today's considerations.

Reply:
I went ahead and removed the sentence.